packages/ckeditor5-core/src/editor/editorui.js

```diff
@@ -172,6 +179,46 @@ export default class EditorUI {
 		return this._editableElementsMap;
 	}
 
+	/**
+	 * Returns viewport offsets object
+	 * { top: Number, right: Number, bottom: Number, left: Number }
+	 * Only top property is currently supported
+	 *
+	 * @private
+	 * @return {Object}
+	 */
+	get _viewportOffset() {
+		const editor = this.editor;
+		const viewportOffsetConfig = editor.config.get( 'ui.viewportOffset' );
+
+		if ( viewportOffsetConfig ) {
+			return viewportOffsetConfig;
+		}
+
+		const legacyOffsetConfig = editor.config.get( 'toolbar.viewportTopOffset' );
+
+		// Fall back to deprecated toolbar config
+		if ( legacyOffsetConfig ) {
```

Comment:
OK, I have very mixed feelings seeing a warning in a `getFoo()` place (a getter for now, but can be a function too). It's just not part of the "reading things logic" to me. However, this is a deprecation notice and after giving it more thought I'm ok after having it here.